src/main/java/com/faforever/client/player/PlayerNoteService.java

```diff
@@ -0,0 +1,99 @@
+package com.faforever.client.player;
+
+import com.faforever.client.domain.PlayerBean;
+import com.faforever.client.preferences.PreferencesService;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.eventbus.EventBus;
+import com.google.common.eventbus.Subscribe;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.beans.factory.InitializingBean;
+import org.springframework.stereotype.Component;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionException;
+
+@Component
+@RequiredArgsConstructor
+@Slf4j
+public class PlayerNoteService implements InitializingBean {
+
+  private static final String PLAYER_NOTES_FILE_NAME = "playerNotes.json";
+
+  private final PreferencesService preferencesService;
+  private final ObjectMapper objectMapper;
+  private final EventBus eventBus;
+
+  private Path notesPath;
+  private Map<Integer, String> notes;
+
+  @Override
+  public void afterPropertiesSet() throws Exception {
+    eventBus.register(this);
+    notesPath = preferencesService.getPreferencesDirectory().resolve(PLAYER_NOTES_FILE_NAME);
```

Comment:
I think it might make more sense to just store the notes in the preferences object itself as an observable map, then you can also add a listener onto the map to add the note to the player. I don't see any advantages to having it in a separate file.

Reply:
Ok, I will try.